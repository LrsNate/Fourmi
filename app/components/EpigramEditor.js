import PropTypes from "prop-types";
import React from "react";

import FourmiPropTypes from "../constants/types";

const EpigramEditor = ({ epigram, onSave }) => {
  return (
    <div>
      <p>{epigram.title}</p>
      <button type="button" onClick={onSave}>
        Go back
      </button>
    </div>
  );
};

EpigramEditor.propTypes = {
  epigram: FourmiPropTypes.epigram.isRequired,
  onSave: PropTypes.func.isRequired
};

export default EpigramEditor;
