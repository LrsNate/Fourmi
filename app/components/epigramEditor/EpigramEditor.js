import PropTypes from "prop-types";
import React from "react";
import { Form } from "react-final-form";

import FourmiPropTypes from "../../constants/types";
import TextInput from "../forms/TextInput";

const AttributesEditor = () => {
  return (
    <div>
      <TextInput name="title" />
      <TextInput name="author" />
      <TextInput name="reference" />
    </div>
  );
};

const EpigramEditor = ({ epigram, onSave }) => {
  /**
   * Cards:
   * 1. Info
   * 2. (Latin Text?)
   * 3. French Text
   * 4. Notes
   */
  return (
    <Form initialValues={epigram} onSubmit={onSave}>
      {({ handleSubmit, values }) => (
        <form onSubmit={handleSubmit}>
          <AttributesEditor />
          <button type="button" onClick={() => onSave(values)}>
            Go back
          </button>
        </form>
      )}
    </Form>
  );
};

EpigramEditor.propTypes = {
  epigram: FourmiPropTypes.epigram.isRequired,
  onSave: PropTypes.func.isRequired
};

export default EpigramEditor;
