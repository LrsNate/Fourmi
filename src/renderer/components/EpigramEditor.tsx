import { Card, CardContent, CardHeader, Grid } from "@material-ui/core";
import * as React from "react";
import { Form } from "react-final-form";
import { Epigram } from "../constants/types";
import TextInput from "./forms/TextInput";

interface EpigramEditorProps {
  epigram: Epigram;
  onSave: (values: Epigram) => void;
}

class EpigramEditor extends React.Component<EpigramEditorProps> {
  public handleSave = (values: object) => {
    this.props.onSave(values as Epigram);
  };

  public render() {
    const { epigram } = this.props;

    return (
      <Form initialValues={epigram} onSubmit={this.handleSave}>
        {({ handleSubmit }) => (
          <form onSubmit={handleSubmit}>{this.renderAttributes()}</form>
        )}
      </Form>
    );
  }

  public renderAttributes() {
    return (
      <Card className={"?"}>
        <CardHeader title="Propriétés" />
        <CardContent>
          <Grid container={true}>
            <Grid item={true} sm={3}>
              <TextInput label="Auteur" name="author" />
            </Grid>
            <Grid item={true} sm={3}>
              <TextInput label="Référence" name="reference" />
            </Grid>
            <Grid item={true} sm={6}>
              <TextInput label="Titre" name="title" />
            </Grid>
            <Grid item={true} sm={3}>
              <TextInput label="Mètre" name="meter" />
            </Grid>
            <Grid item={true} sm={3}>
              <TextInput label="Strophe" name="stanza" />
            </Grid>
            <Grid item={true} sm={3}>
              <TextInput label="Destinataire" name="addressee" />
            </Grid>
            <Grid item={true} sm={3}>
              <TextInput label="Date" name="date" />
            </Grid>
            <Grid item={true} sm={12}>
              <TextInput label="Thèmes" name="themes" />
            </Grid>
          </Grid>
        </CardContent>
      </Card>
    );
  }
}

export default EpigramEditor;
