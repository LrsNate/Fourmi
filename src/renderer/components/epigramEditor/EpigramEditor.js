import {
  Button,
  Card,
  CardContent,
  CardHeader,
  Grid,
  withStyles
} from "material-ui";
import { Save } from "material-ui-icons";
import PropTypes from "prop-types";
import React, { Component } from "react";
import { Form } from "react-final-form";
import { Link } from "react-router-dom";
import { selectRoute } from "../../constants/routes";

import FourmiPropTypes from "../../constants/types";
import TextInput from "../forms/TextInput";
import TextEditor from "../forms/TextEditor";

const styles = theme => ({
  card: {
    marginBottom: theme.spacing.unit
  },
  saveButton: {
    position: "fixed",
    right: theme.spacing.unit * 2,
    bottom: theme.spacing.unit * 2,
    zIndex: 1
  }
});

class EpigramEditor extends Component {
  static propTypes = {
    classes: PropTypes.object,
    epigram: FourmiPropTypes.epigram.isRequired,
    goToSelectPage: PropTypes.func.isRequired,
    onSave: PropTypes.func.isRequired
  };

  static defaultProps = {
    classes: {}
  };

  renderAttributes() {
    const { classes, goToSelectPage } = this.props;

    return (
      <Card className={classes.card}>
        <CardHeader title="Propriétés" />
        <CardContent>
          <Grid container>
            <Grid item sm={3}>
              <TextInput label="Auteur" name="author" />
            </Grid>
            <Grid item sm={3}>
              <TextInput label="Référence" name="reference" />
            </Grid>
            <Grid item sm={6}>
              <TextInput label="Titre" name="title" />
            </Grid>
            <Grid item sm={3}>
              <TextInput label="Mètre" name="meter" />
            </Grid>
            <Grid item sm={3}>
              <TextInput label="Strophe" name="stanza" />
            </Grid>
            <Grid item sm={3}>
              <TextInput label="Destinataire" name="addressee" />
            </Grid>
            <Grid item sm={3}>
              <TextInput label="Date" name="date" />
            </Grid>
            <Grid item sm={12}>
              <TextInput label="Vices" name="vices" />
            </Grid>
          </Grid>
          <Grid item sm={12}>
            <Button onClick={goToSelectPage}>Origine</Button>
          </Grid>
        </CardContent>
      </Card>
    );
  }

  renderTextEditor(label, name) {
    const { classes } = this.props;

    return (
      <Card className={classes.card}>
        <CardHeader title={label} />
        <CardContent>
          <TextEditor name={name} />
        </CardContent>
      </Card>
    );
  }

  render() {
    const { classes, epigram, onSave } = this.props;
    return (
      <Form initialValues={epigram} onSubmit={onSave}>
        {({ handleSubmit }) => (
          <form onSubmit={handleSubmit}>
            {this.renderAttributes()}
            {this.renderTextEditor("Texte latin", "latinText")}
            {this.renderTextEditor("Texte français", "frenchText")}
            {this.renderTextEditor("Notes", "notes")}
            <Button
              className={classes.saveButton}
              type="submit"
              fab
              color="accent"
            >
              <Save />
            </Button>
          </form>
        )}
      </Form>
    );
  }
}

export default withStyles(styles)(EpigramEditor);
