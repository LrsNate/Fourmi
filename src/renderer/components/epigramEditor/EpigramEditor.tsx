import {
  Button,
  Card,
  CardContent,
  CardHeader,
  Grid,
  StyleRulesCallback,
  Theme,
  withStyles
} from "@material-ui/core";
import { Save } from "@material-ui/icons";
import * as React from "react";
import { Field, Form } from "react-final-form";
import { RouteComponentProps, withRouter } from "react-router";
import { Epigram } from "../../constants/types";
import { searchRoutePath } from "../../routes";
import TextEditor from "../forms/TextEditor";
import TextInput from "../forms/TextInput";
import ChipEditorCard from "./ChipEditorCard";

const styles: StyleRulesCallback<string> = (theme: Theme) => ({
  saveButton: {
    position: "fixed",
    right: theme.spacing.unit * 2,
    bottom: theme.spacing.unit * 2,
    zIndex: 1
  }
});

interface EpigramEditorProps extends RouteComponentProps<{}> {
  epigram: Epigram;
  onSave: (values: Epigram) => void;
  classes: Record<string, string>;
}

class EpigramEditor extends React.Component<EpigramEditorProps> {
  public handleSave = (values: object) => {
    const { history, onSave } = this.props;
    onSave(values as Epigram);
    history.push(searchRoutePath());
  };

  public render() {
    const { classes, epigram } = this.props;

    return (
      <Form initialValues={epigram} onSubmit={this.handleSave}>
        {({ handleSubmit }) => (
          <form onSubmit={handleSubmit}>
            <Grid container direction="column" spacing={8}>
              <Grid item>{this.renderAttributes()}</Grid>
              <Grid item>{this.renderThemes()}</Grid>
              <Grid item>
                {this.renderTextEditor("Texte latin", "latinText")}
              </Grid>
              <Grid item>
                {this.renderTextEditor("Texte français", "frenchText")}
              </Grid>
              <Grid item>{this.renderTextEditor("Notes", "notes")}</Grid>
            </Grid>
            <Button
              className={classes.saveButton}
              type="submit"
              variant="fab"
              color="secondary"
            >
              <Save />
            </Button>
          </form>
        )}
      </Form>
    );
  }

  public renderAttributes() {
    return (
      <Card>
        <CardHeader title="Propriétés" />
        <CardContent>
          <Grid container spacing={8}>
            <Grid item sm={3}>
              <TextInput label="Auteur" name="author" />
            </Grid>
            <Grid item sm={3}>
              <TextInput label="Référence" name="reference" />
            </Grid>
            <Grid item sm={6}>
              <TextInput label="Titre" name="title" />
            </Grid>
            <Grid item sm={3}>
              <TextInput label="Mètre" name="meter" />
            </Grid>
            <Grid item sm={3}>
              <TextInput label="Strophe" name="stanza" />
            </Grid>
            <Grid item sm={3}>
              <TextInput label="Destinataire" name="addressee" />
            </Grid>
            <Grid item sm={3}>
              <TextInput label="Date" name="date" />
            </Grid>
          </Grid>
        </CardContent>
      </Card>
    );
  }

  public renderThemes() {
    return (
      <Field name="themes">
        {({ input: { onChange, value } }) => (
          <ChipEditorCard values={value} onChange={onChange} />
        )}
      </Field>
    );
  }

  public renderTextEditor(label: string, name: string) {
    return (
      <Card>
        <CardHeader title={label} />
        <CardContent>
          <TextEditor name={name} />
        </CardContent>
      </Card>
    );
  }
}

export default withStyles(styles)(withRouter(EpigramEditor));
