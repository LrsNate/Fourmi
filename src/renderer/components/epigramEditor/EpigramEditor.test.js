import { render } from "enzyme";
import React from "react";

import EpigramEditor from "./EpigramEditor";

describe("The EpigramEditor component", () => {
  describe("on an initial load", () => {
    it("matches the snapshot", () => {
      const epigram = {};
      const wrapper = render(
        <EpigramEditor
          epigram={epigram}
          onSave={() => {}}
          goToSelectPage={() => {}}
        />
      );

      expect(wrapper).toMatchSnapshot();
    });
  });
});
