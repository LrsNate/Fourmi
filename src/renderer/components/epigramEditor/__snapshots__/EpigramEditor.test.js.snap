// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The EpigramEditor component on an initial load matches the snapshot 1`] = `
<form>
  <div
    class="MuiPaper-root-3 MuiPaper-shadow2-7 MuiPaper-rounded-4 EpigramEditor-card-1"
  >
    <div
      class="MuiCardContent-root-36 MuiCardHeader-root-30"
    >
      <div
        class="MuiCardHeader-content-33"
      >
        <span
          class="MuiTypography-root-37 MuiTypography-headline-42 MuiCardHeader-title-34"
        >
          Propriétés
        </span>
      </div>
    </div>
    <div
      class="MuiCardContent-root-36"
    >
      <div
        class="MuiGrid-typeContainer-61 MuiGrid-spacing-xs-16-82"
      >
        <div
          class="MuiGrid-typeItem-62 MuiGrid-grid-sm-3-101"
        >
          <div
            class="MuiFormControl-root-150 MuiFormControl-marginNormal-151 MuiFormControl-fullWidth-153"
          >
            <label
              class="MuiFormLabel-root-160 MuiInputLabel-root-154 MuiInputLabel-formControl-155 MuiInputLabel-animated-158"
              data-shrink="false"
            >
              Auteur
            </label>
            <div
              class="MuiInput-root-164 MuiInput-formControl-165 MuiInput-inkbar-166 MuiInput-underline-170"
            >
              <input
                aria-invalid="false"
                aria-required="false"
                class="MuiInput-input-173 MuiInput-inputSingleline-176"
                name="author"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-typeItem-62 MuiGrid-grid-sm-3-101"
        >
          <div
            class="MuiFormControl-root-150 MuiFormControl-marginNormal-151 MuiFormControl-fullWidth-153"
          >
            <label
              class="MuiFormLabel-root-160 MuiInputLabel-root-154 MuiInputLabel-formControl-155 MuiInputLabel-animated-158"
              data-shrink="false"
            >
              Référence
            </label>
            <div
              class="MuiInput-root-164 MuiInput-formControl-165 MuiInput-inkbar-166 MuiInput-underline-170"
            >
              <input
                aria-invalid="false"
                aria-required="false"
                class="MuiInput-input-173 MuiInput-inputSingleline-176"
                name="reference"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-typeItem-62 MuiGrid-grid-sm-6-104"
        >
          <div
            class="MuiFormControl-root-150 MuiFormControl-marginNormal-151 MuiFormControl-fullWidth-153"
          >
            <label
              class="MuiFormLabel-root-160 MuiInputLabel-root-154 MuiInputLabel-formControl-155 MuiInputLabel-animated-158"
              data-shrink="false"
            >
              Titre
            </label>
            <div
              class="MuiInput-root-164 MuiInput-formControl-165 MuiInput-inkbar-166 MuiInput-underline-170"
            >
              <input
                aria-invalid="false"
                aria-required="false"
                class="MuiInput-input-173 MuiInput-inputSingleline-176"
                name="title"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-typeItem-62 MuiGrid-grid-sm-3-101"
        >
          <div
            class="MuiFormControl-root-150 MuiFormControl-marginNormal-151 MuiFormControl-fullWidth-153"
          >
            <label
              class="MuiFormLabel-root-160 MuiInputLabel-root-154 MuiInputLabel-formControl-155 MuiInputLabel-animated-158"
              data-shrink="false"
            >
              Mètre
            </label>
            <div
              class="MuiInput-root-164 MuiInput-formControl-165 MuiInput-inkbar-166 MuiInput-underline-170"
            >
              <input
                aria-invalid="false"
                aria-required="false"
                class="MuiInput-input-173 MuiInput-inputSingleline-176"
                name="meter"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-typeItem-62 MuiGrid-grid-sm-3-101"
        >
          <div
            class="MuiFormControl-root-150 MuiFormControl-marginNormal-151 MuiFormControl-fullWidth-153"
          >
            <label
              class="MuiFormLabel-root-160 MuiInputLabel-root-154 MuiInputLabel-formControl-155 MuiInputLabel-animated-158"
              data-shrink="false"
            >
              Strophe
            </label>
            <div
              class="MuiInput-root-164 MuiInput-formControl-165 MuiInput-inkbar-166 MuiInput-underline-170"
            >
              <input
                aria-invalid="false"
                aria-required="false"
                class="MuiInput-input-173 MuiInput-inputSingleline-176"
                name="stanza"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-typeItem-62 MuiGrid-grid-sm-3-101"
        >
          <div
            class="MuiFormControl-root-150 MuiFormControl-marginNormal-151 MuiFormControl-fullWidth-153"
          >
            <label
              class="MuiFormLabel-root-160 MuiInputLabel-root-154 MuiInputLabel-formControl-155 MuiInputLabel-animated-158"
              data-shrink="false"
            >
              Destinataire
            </label>
            <div
              class="MuiInput-root-164 MuiInput-formControl-165 MuiInput-inkbar-166 MuiInput-underline-170"
            >
              <input
                aria-invalid="false"
                aria-required="false"
                class="MuiInput-input-173 MuiInput-inputSingleline-176"
                name="addressee"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-typeItem-62 MuiGrid-grid-sm-3-101"
        >
          <div
            class="MuiFormControl-root-150 MuiFormControl-marginNormal-151 MuiFormControl-fullWidth-153"
          >
            <label
              class="MuiFormLabel-root-160 MuiInputLabel-root-154 MuiInputLabel-formControl-155 MuiInputLabel-animated-158"
              data-shrink="false"
            >
              Date
            </label>
            <div
              class="MuiInput-root-164 MuiInput-formControl-165 MuiInput-inkbar-166 MuiInput-underline-170"
            >
              <input
                aria-invalid="false"
                aria-required="false"
                class="MuiInput-input-173 MuiInput-inputSingleline-176"
                name="date"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-typeItem-62 MuiGrid-grid-sm-12-110"
        >
          <div
            class="MuiFormControl-root-150 MuiFormControl-marginNormal-151 MuiFormControl-fullWidth-153"
          >
            <label
              class="MuiFormLabel-root-160 MuiInputLabel-root-154 MuiInputLabel-formControl-155 MuiInputLabel-animated-158"
              data-shrink="false"
            >
              Vices
            </label>
            <div
              class="MuiInput-root-164 MuiInput-formControl-165 MuiInput-inkbar-166 MuiInput-underline-170"
            >
              <input
                aria-invalid="false"
                aria-required="false"
                class="MuiInput-input-173 MuiInput-inputSingleline-176"
                name="vices"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-typeItem-62 MuiGrid-grid-sm-12-110"
      >
        <button
          class="MuiButtonBase-root-194 MuiButton-root-179"
          role="button"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label-181"
          >
            Origine
          </span>
          <span
            class="MuiTouchRipple-root-196"
          />
        </button>
      </div>
    </div>
  </div>
  <div
    class="MuiPaper-root-3 MuiPaper-shadow2-7 MuiPaper-rounded-4 EpigramEditor-card-1"
  >
    <div
      class="MuiCardContent-root-36 MuiCardHeader-root-30"
    >
      <div
        class="MuiCardHeader-content-33"
      >
        <span
          class="MuiTypography-root-37 MuiTypography-headline-42 MuiCardHeader-title-34"
        >
          Texte latin
        </span>
      </div>
    </div>
    <div
      class="MuiCardContent-root-36"
    >
      <div>
        Rich text editor mock
      </div>
    </div>
  </div>
  <div
    class="MuiPaper-root-3 MuiPaper-shadow2-7 MuiPaper-rounded-4 EpigramEditor-card-1"
  >
    <div
      class="MuiCardContent-root-36 MuiCardHeader-root-30"
    >
      <div
        class="MuiCardHeader-content-33"
      >
        <span
          class="MuiTypography-root-37 MuiTypography-headline-42 MuiCardHeader-title-34"
        >
          Texte français
        </span>
      </div>
    </div>
    <div
      class="MuiCardContent-root-36"
    >
      <div>
        Rich text editor mock
      </div>
    </div>
  </div>
  <div
    class="MuiPaper-root-3 MuiPaper-shadow2-7 MuiPaper-rounded-4 EpigramEditor-card-1"
  >
    <div
      class="MuiCardContent-root-36 MuiCardHeader-root-30"
    >
      <div
        class="MuiCardHeader-content-33"
      >
        <span
          class="MuiTypography-root-37 MuiTypography-headline-42 MuiCardHeader-title-34"
        >
          Notes
        </span>
      </div>
    </div>
    <div
      class="MuiCardContent-root-36"
    >
      <div>
        Rich text editor mock
      </div>
    </div>
  </div>
  <button
    class="MuiButtonBase-root-194 MuiButton-root-179 MuiButton-raised-186 MuiButton-fab-192 MuiButton-raisedAccent-189 EpigramEditor-saveButton-2"
    role="button"
    tabindex="0"
    type="submit"
  >
    <span
      class="MuiButton-label-181"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root-205"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M17 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V7l-4-4zm-5 16c-1.66 0-3-1.34-3-3s1.34-3 3-3 3 1.34 3 3-1.34 3-3 3zm3-10H5V5h10v4z"
        />
      </svg>
    </span>
    <span
      class="MuiTouchRipple-root-196"
    />
  </button>
</form>
`;
